peer_heap.go

```diff
@@ -44,7 +45,7 @@ func (ph *peerHeap) Less(i, j int) bool {
 	if ph.peerScores[i].score == ph.peerScores[j].score {
 		return ph.peerScores[i].order < ph.peerScores[j].order
```

Comment:
order is now an `atomic.Uint64` so it should fail to compile right?

also need to figure out what we want to do here -- this code path needs to be very fast since we have a lot of peers. there's no atomics here currently, I think that's because the caller of the peer heap locks and so it might be safe not to. Let's look into this a little more.